.github/actions/is-release-week-v1/src/utils.ts

```diff
@@ -0,0 +1,17 @@
+export function getWeekNumber(date: Date): number {
+  const firstDayOfYear: Date = new Date(date.getTime())
+  firstDayOfYear.setMonth(0, 1)
+  const dayOfYear =
+    Math.floor(
+      (date.getTime() - firstDayOfYear.getTime()) / (24 * 60 * 60 * 1000)
+    ) + 1
+  const weekNumber = Math.ceil((firstDayOfYear.getDay() + dayOfYear) / 7)
+
+  return weekNumber
```

Comment:
This algorithm isn't quite correct. When asked what the week number is for [Jan 1, 1863](https://luirig.altervista.org/calendar/index.php?year=1863), it returns `0`, and for Jan 4 it returns `1` (should be the 2nd week of the year). Looks like it just counts the days in between the first day of the year and the current date, but doesn't factor in how far into the week the first of the year started.

[This algorithm](https://javascript.plainenglish.io/how-to-get-the-week-of-year-of-a-given-date-in-javascript-c8fe0d764b5d) seems to produce the correct output (testing it further March 30 and April 2 are both week `14`).

```suggestion
  const now = new Date(2021, 3, 1);
  const onejan = new Date(date.getFullYear(), 0, 1);
  const week = Math.ceil((((date.getTime() - onejan.getTime()) / 
  86400000) + onejan.getDay() + 1) / 7);

  return week
```

Reply:
The current algorithm is the same as the [the one](https://javascript.plainenglish.io/how-to-get-the-week-of-year-of-a-given-date-in-javascript-c8fe0d764b5d) you referred.

It returns as following:
```
  assert.isTrue(getWeekNumber(new Date(1863, 0, 1)) === 1)
  assert.isTrue(getWeekNumber(new Date(1863, 0, 4)) === 2)
```

Reply:
Oh, you're right. Turns out I was using a different algorithm thinking it was yours. :P